### PR TITLE
Added SDK for Region Move.

### PR DESCRIPTION
**Pull request by ajaygit158**
Added a new SDK for region move service API which helps move resources from one region to another.

### CODE REVIEW

**Comment by allenjzhang** [comment]
One question, I see that in rest-api-specs repo, there are `regionmove` AND `migrate` both to Microsoft.Migrate provider name. Are they the same? Per convention, shouldn't the folder name .net and in res-api-specs be **\migrate**? 

Also given this is new, the check pipeline may not be set up correctly. I will get back to you if any thing is needed from you. Thanks.

**Comment by ajaygit158** [comment]
@allenjzhang 
We assumed this is allowed as we could see similar structure for recoveryservices-backup and siterecovery and both have different folders. Also, we believe its cleaner to keep it separate as the tow are different services, even though they share the same RP

**Comment by ajaygit158** [comment]
@allenjzhang One more question. In our swagger, for the MoveResourceProperties object, we have marked certain properties as readOnly. Accordingly, the generated SDK has private setters for these fields. However, the parameterized constructor still allows setting these properties. Do we have to handle this via CustomInit() manually?

eg:   
`public MoveResourcePropertiesMoveStatus MoveStatus { get; private set; }`

```
public MoveResourceProperties(string sourceId, string provisioningState = default(string), string targetId = default(string), string existingTargetId = default(string), ResourceSettings resourceSettings = default(ResourceSettings), MoveResourcePropertiesSourceResourceSettings sourceResourceSettings = default(MoveResourcePropertiesSourceResourceSettings), MoveResourcePropertiesMoveStatus moveStatus = default(MoveResourcePropertiesMoveStatus), IList<MoveResourceDependency> dependsOn = default(IList<MoveResourceDependency>), IList<MoveResourceDependencyOverride> dependsOnOverrides = default(IList<MoveResourceDependencyOverride>), MoveResourceError errors = default(MoveResourceError))
{
            ProvisioningState = provisioningState;
            SourceId = sourceId;
            TargetId = targetId;
            ExistingTargetId = existingTargetId;
            ResourceSettings = resourceSettings;
            SourceResourceSettings = sourceResourceSettings;
            MoveStatus = moveStatus;
            DependsOn = dependsOn;
            DependsOnOverrides = dependsOnOverrides;
            Errors = errors;
            CustomInit();
        }
```

**Review by allenjzhang — CHANGES_REQUESTED** [review]
I am fixing the pipeline to make sure CI runs correctly. One change I would like to request is can you add assemblyinfo.cs and make sure the `AssemblyFileVersion` and `AssemblyVersion` matches the package version?

**Review by allenjzhang — APPROVED** [review]

File: sdk/regionmove/Microsoft.Azure.Management.Migrate/AzSdk.RP.props [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <!--This file and it's contents are updated at build time moving or editing might result in build failure. Take due deligence while editing this file-->
+  <PropertyGroup>
+    <AzureApiTag>AzureResourceMover_2019-10-01-preview;</AzureApiTag>
+    <PackageTags>$(PackageTags);$(CommonTags);$(AzureApiTag);</PackageTags>
+  </PropertyGroup>
+</Project>
```
Comment (allenjzhang) [inline sdk/regionmove/Microsoft.Azure.Management.Migrate/AzSdk.RP.props:7]: Can you do me a favor and make another commit to add a line return here? I merged the pipeline change after your commit. Running `check-enforcer /reset` didn't seem to retrigger it. Another commit may help run it normally. Thanks

Comment (ajaygit158) [reply sdk/regionmove/Microsoft.Azure.Management.Migrate/AzSdk.RP.props]: Done

**Comment by allenjzhang** [comment]
> @allenjzhang One more question. In our swagger, for the MoveResourceProperties object, we have marked certain properties as readOnly. Accordingly, the generated SDK has private setters for these fields. However, the parameterized constructor still allows setting these properties. Do we have to handle this via CustomInit() manually?
> 
> eg:
> `public MoveResourcePropertiesMoveStatus MoveStatus { get; private set; }`
> 
> ```
> public MoveResourceProperties(string sourceId, string provisioningState = default(string), string targetId = default(string), string existingTargetId = default(string), ResourceSettings resourceSettings = default(ResourceSettings), MoveResourcePropertiesSourceResourceSettings sourceResourceSettings = default(MoveResourcePropertiesSourceResourceSettings), MoveResourcePropertiesMoveStatus moveStatus = default(MoveResourcePropertiesMoveStatus), IList<MoveResourceDependency> dependsOn = default(IList<MoveResourceDependency>), IList<MoveResourceDependencyOverride> dependsOnOverrides = default(IList<MoveResourceDependencyOverride>), MoveResourceError errors = default(MoveResourceError))
> {
>             ProvisioningState = provisioningState;
>             SourceId = sourceId;
>             TargetId = targetId;
>             ExistingTargetId = existingTargetId;
>             ResourceSettings = resourceSettings;
>             SourceResourceSettings = sourceResourceSettings;
>             MoveStatus = moveStatus;
>             DependsOn = dependsOn;
>             DependsOnOverrides = dependsOnOverrides;
>             Errors = errors;
>             CustomInit();
>         }
> ```

This should be fine. It basically allows the object construction but prevents reassignment.
